tests/bluetooth/controller/ctrl_feature_exchange/src/main.c

```diff
@@ -142,83 +151,154 @@ void test_feature_exchange_mas_loc_2(void)
-	sys_put_le64(featureset, local_feature_rsp.features);
-
 	test_set_role(&conn, BT_HCI_ROLE_MASTER);
+	/* Connect */
 	ull_cp_state_set(&conn, ULL_CP_CONNECTED);
 
-	event_prepare(&conn);
+	for (int feat_count = 0; feat_count < feat_to_test; feat_count++) {
+		sys_put_le64(set_featureset[feat_count],
+			     remote_feature_req.features);
+		sys_put_le64(exp_featureset[feat_count],
+			     local_feature_rsp.features);
 
-	lt_tx(LL_SLAVE_FEATURE_REQ, &conn, &remote_feature_req);
+		event_prepare(&conn);
 
-	event_done(&conn);
+		lt_tx(LL_SLAVE_FEATURE_REQ, &conn, &remote_feature_req);
 
-	event_prepare(&conn);
+		event_done(&conn);
 
-	lt_rx(LL_FEATURE_RSP, &conn, &tx, &local_feature_rsp);
-	lt_rx_q_is_empty(&conn);
+		event_prepare(&conn);
 
-	event_done(&conn);
+		lt_rx(LL_FEATURE_RSP, &conn, &tx, &local_feature_rsp);
+		lt_rx_q_is_empty(&conn);
 
-	ut_rx_q_is_empty();
-	zassert_equal(conn.lll.event_counter, 2,
+		event_done(&conn);
+
+		ut_rx_q_is_empty();
+
+		ull_cp_release_tx(tx);
+	}
+	zassert_equal(conn.lll.event_counter,
+		      MAS_REM_NR_OF_EVENTS*(feat_to_test),
 		      "Wrong event-count %d\n", conn.lll.event_counter);
 }
 
+#define MAS_REM_2_NR_OF_EVENTS 3
 void test_feature_exchange_mas_rem_2(void)
 {
-	u64_t featureset;
+	/*
+	 * we could combine some of the following,
+	 * but in reality we should add some more
+	 * test cases
+	 */
+	u64_t set_featureset[] = {
+		DEFAULT_FEATURE,
+		LL_FEAT_BIT_MASK_VALID,
+		EXPECTED_FEAT_EXCH_VALID,
+		0xFFFFFFFFFFFFFFFF,
+		0x0 };
+	u64_t exp_featureset[] = {
+		DEFAULT_FEATURE,
+		DEFAULT_FEATURE,
+		DEFAULT_FEATURE,
+		DEFAULT_FEATURE,
+		0x0 };
+	u64_t ut_featureset[] = {
+		DEFAULT_FEATURE,
+		DEFAULT_FEATURE,
+		DEFAULT_FEATURE,
+		DEFAULT_FEATURE,
+		DEFAULT_FEATURE };
+	u64_t ut_exp_featureset[] = {
+		DEFAULT_FEATURE,
+		DEFAULT_FEATURE,
+		DEFAULT_FEATURE,
+		DEFAULT_FEATURE,
+		0x00 };
+
+	int feat_to_test = ARRAY_SIZE(set_featureset);
 	u64_t err;
 	struct node_tx *tx;
 	struct node_rx_pdu *ntf;
 
 	struct pdu_data_llctrl_feature_req remote_feature_req;
 	struct pdu_data_llctrl_feature_rsp local_feature_rsp;
-
-	featureset = DEFAULT_FEATURE;
-
-	sys_put_le64(featureset, remote_feature_req.features);
-	sys_put_le64(featureset, local_feature_rsp.features);
+	struct pdu_data_llctrl_feature_req ut_feature_req;
+	struct pdu_data_llctrl_feature_req ut_feature_rsp;
 
 	test_set_role(&conn, BT_HCI_ROLE_MASTER);
 	ull_cp_state_set(&conn, ULL_CP_CONNECTED);
 
+	for (int feat_count = 0; feat_count < feat_to_test; feat_count++) {
+		sys_put_le64(set_featureset[feat_count],
+			     remote_feature_req.features);
+		sys_put_le64(exp_featureset[feat_count],
+			     local_feature_rsp.features);
+		sys_put_le64(ut_featureset[feat_count],
+			     ut_feature_req.features);
+		sys_put_le64(ut_exp_featureset[feat_count],
+			     ut_feature_rsp.features);
+
+		/*
+		 * at the start of a loop all queues should be empty
+		 */
+		ut_rx_q_is_empty();
```

Comment:
These checks could be removed since the queues are implicitly empty @ start and confirmed empty @ loop-end